hoodie-client/src/main/java/com/uber/hoodie/table/HoodieMergeOnReadTable.java

```diff
@@ -216,11 +217,16 @@ public Partitioner getUpsertPartitioner(WorkloadProfile profile) {
                   final Map<FileStatus, Boolean> filesToDeletedStatus = new HashMap<>();
                   Map<FileStatus, Long> filesToNumBlocksRollback = new HashMap<>();
 
-                  // In case all data was inserts and the commit failed, there is no partition stats
-                  if (commitMetadata.getPartitionToWriteStats().size() == 0) {
-                    super.deleteCleanedFiles(filesToDeletedStatus, partitionPath, filter);
-                  }
-
+                  // In case all data was inserts and the commit failed, delete the file belonging to that commit
+                  super.deleteCleanedFiles(filesToDeletedStatus, partitionPath, filter);
+                  final Set<String> deletedFiles = filesToDeletedStatus.entrySet().stream()
+                      .map(entry -> {
+                        Path filePath = entry.getKey().getPath();
```

Comment:
Minor : Looks like a reusable functionality of finding the fileId from a file-path (either log-file or parquet). Can you introduce a FSUtils method and move the logic there ?

Reply:
done